prime-router/src/test/kotlin/fhirengine/azure/FhirFunctionTests.kt

```diff
@@ -732,30 +802,162 @@ class FhirFunctionTests {
+
+            every {
+                BlobAccess.Companion.uploadBody(
+                    any(),
+                    any(),
+                    any(),
+                    any(),
+                    any()
+                )
+            } returns BlobAccess.BlobInfo(Report.Format.FHIR, "", "".toByteArray())
+            every { queueMock.sendMessage(any(), any()) } returns Unit
+
+            val settings = FileSettings().loadOrganizations(oneOrganization)
+            val fhirEngine = spyk(
+                FHIRTranslator(
+                    UnitTestUtils.simpleMetadata,
+                    settings,
+                    ReportStreamTestDatabaseContainer.testDatabaseAccess,
+                    blobMock,
+                )
+            )
+
+            val actionHistory = spyk(ActionHistory(TaskAction.receive))
+            val workflowEngine =
+                makeWorkflowEngine(
+                    UnitTestUtils.simpleMetadata,
+                    settings,
+                    ReportStreamTestDatabaseContainer.testDatabaseAccess
+                )
+
+            val queueMessage = "{\"type\":\"translate\",\"reportId\":\"${translateReport.id}\"," +
+                "\"blobURL\":\"http://azurite:10000/devstoreaccount1/reports/receive%2Fignore.ignore-full-elr%2F" +
+                "None-${translateReport.id}.hl7\",\"digest\":\"${
+                    BlobAccess.digestToString(
+                        BlobAccess.sha256Digest(
+                            translateFhirBytes
+                        )
+                    )
+                }\",\"blobSubFolderName\":" +
+                "\"ignore.ignore-full-elr\",\"schemaName\":\"\",\"topic\":\"elr-elims\"," +
+                "\"receiverFullName\":\"phd.elr2\"}"
+
+            val fhirFunc = FHIRFunctions(
+                workflowEngine,
+                queueAccess = queueMock,
+                databaseAccess = ReportStreamTestDatabaseContainer.testDatabaseAccess
+            )
+
+            fhirFunc.doTranslate(queueMessage, 1, fhirEngine, actionHistory)
+
+            // verify task and report_file tables were updated correctly in the Translate function
+            ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+                val sendTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
+                    .where(Task.TASK.NEXT_ACTION.eq(TaskAction.send))
+                    .fetchOneInto(Task.TASK)
+                assertThat(sendTask).isNotNull()
+
+                val sendReportFile =
+                    DSL.using(txn).select(REPORT_FILE.asterisk())
+                        .from(REPORT_FILE)
+                        .where(REPORT_FILE.REPORT_ID.eq(sendTask!!.reportId))
+                        .fetchOneInto(REPORT_FILE)
+                assertThat(sendReportFile).isNotNull()
+            }
+            // verify we downloaded the original report in the Translate step
+            verify(exactly = 1) {
```

Comment:
Could we add an additional verification here that we uploaded to the blob store with the correct file contents?

Reply:
Will revisit this in #13232 and also add smokeTests that verify contents.